### ISSUE TITLE
How to add the scheduler to the Model CLIP?

### ISSUE THREAD

**Issue by APiaoG** [openai/CLIP#332]
I tried to add a scheduler to the model CLIP, but the learning rate didn't update. I'm sure I add the WarmupCosineSchedule and scheduler.step(). so what is the problem of it?

**Comment by MadcowD** [comment]
Have you tried using the StepLR scheduler instead of the
WarmupCosineSchedule? It may work better for your needs.

**Comment by APiaoG** [comment]
Thank you and I have solved this problem!

**Comment by MuhiimAli** [comment]
How did you solve the problem? I am running into the same issue now
